axelrod/tests/strategies/test_grudger.py

```diff
@@ -336,3 +336,69 @@ def test_strategy(self):
         opponent = axl.MockPlayer(actions=opponent_actions)
         actions = [(C, D)] * 2 + [(D, D)] * 18 + [(D, C)] * 20
         self.versus_test(opponent, expected_actions=actions)
+
+
+class TestCapri(TestPlayer):
+
+    name = "CAPRI"
+    player = axl.Capri
+    expected_classifier = {
+        "memory_depth": 3,
+        "stochastic": False,
+        "makes_use_of": set(),
+        "long_run_time": False,
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    def assert_prescription(self, player_history, opponent_history, prescription):
+        player = self.player()
+        player._history = player_history
+        opponent = axl.MockPlayer()
+        opponent._history = opponent_history
+        self.assertEqual(player.strategy(opponent=opponent), prescription)
```

Comment:
We have moved away from directly setting the `._history` like this in tests would you be able to write this using the `MockPlayer` and a sequence of actions?

This would have the beneficial side effect of identifying if any pairs of histories cannot happen (for some reason or another).

Reply:
After I struggled for a while, I find it difficult to find a proper seed. Could you help me find a proper seed?
It would be great if we can specify when the actions are flipped instead of specifying `noise` and `seed` in a future version of Axelrod library. For instance, I would like an interface like

```python
self.versus_test(axelrod.Alternator(), expected_actions=actions, flip=[0,0,0,1,0,0,0,0] )
```
to indicate that the action of the focal player is flipped in the fourth round of the match. It will also improve the readability of the test code.

Reply:
I usually use a loop:
```
for seed in range(0, 10000):
  # setup Match with seed
  # check for desired conditions, break and print seed
```

There's a good reason we do things this way -- playing through a `Match` can affect the internal variables of a strategy and lead to subtle bugs.

Reply:
Thanks! Actually, I found a useful method `TestMatch#search_seeds` which does exactly what you suggested.
I rewrote the test cases specifying seeds. See 9c2c8f2d

Reply:
Great! I should have remembered that method, I'm pretty sure I wrote it!